web/src/lib/taxation/helpers.ts

```diff
@@ -0,0 +1,132 @@
+      businessNameToSubmitToTaxApi = props.profileData.responsibleOwnerName;
+    }
+
+    const userDataToSet = await postTaxFilingsOnboarding({
+      taxId: props.profileData.taxId as string,
+      businessName: businessNameToSubmitToTaxApi,
+      encryptedTaxId: encryptedTaxId as string,
+    });
+
+    props.updateQueue.queue(userDataToSet).queueProfileData({
+      taxId: props.profileData.taxId,
+      encryptedTaxId: encryptedTaxId,
+    });
+
+    if (getCurrentBusiness(userDataToSet).taxFilingData.state === "SUCCESS") {
+      if (displayBusinessName(props.business)) {
+        props.updateQueue.queueProfileData({
+          businessName: props.profileData.businessName,
+        });
+      }
+
+      if (displayResponsibleOwnerName(props.business)) {
+        props.updateQueue.queueProfileData({
+          responsibleOwnerName: props.profileData.responsibleOwnerName,
+        });
+      }
+    }
+
+    await props.updateQueue.update();
+  } catch {
+    props.setOnAPIfailed && props.setOnAPIfailed("UNKNOWN");
+    props.setIsLoading && props.setIsLoading(false);
+    return;
+  }
+
+  const { taxFilingData } = props.updateQueue.currentBusiness();
+
+  if (taxFilingData.state === "SUCCESS") {
+    props.setIsLoading && props.setIsLoading(false);
+    analytics.event.tax_calendar_modal.submit.tax_deadlines_added_to_calendar();
+    props.onSuccess && props.onSuccess();
+    props.queueUpdateTaskProgress("determine-naics-code", "COMPLETED");
+    props.updateQueue.update();
+  }
+
+  if (taxFilingData.state === "PENDING") {
+    props.setIsLoading && props.setIsLoading(false);
+    analytics.event.tax_calendar_modal.submit.business_exists_but_not_in_Gov2Go();
+    props.close && props.close();
+  }
+
+  if (taxFilingData.state === "FAILED") {
+    if (taxFilingData.errorField === "businessName" && displayBusinessName(props.business)) {
+      props.formContextState &&
+        props.formContextState.reducer({
+          type: FieldStateActionKind.VALIDATION,
+          payload: { field: "businessName", invalid: true },
+        });
+    } else if (taxFilingData.errorField === "businessName" && displayResponsibleOwnerName(props.business)) {
+      props.formContextState &&
+        props.formContextState.reducer({
+          type: FieldStateActionKind.VALIDATION,
+          payload: { field: "responsibleOwnerName", invalid: true },
+        });
+    } else {
+      props.formContextState &&
+        props.formContextState.reducer({
+          type: FieldStateActionKind.VALIDATION,
+          payload: { field: fields, invalid: true },
+        });
+    }
+    props.setOnAPIfailed && props.setOnAPIfailed("FAILED");
+    analytics.event.tax_calendar_modal.submit.tax_calendar_business_does_not_exist();
+    props.setIsLoading && props.setIsLoading(false);
+  }
+
+  if (taxFilingData.state === "API_ERROR") {
+    props.setOnAPIfailed && props.setOnAPIfailed("UNKNOWN");
+    props.setIsLoading && props.setIsLoading(false);
+  }
+};
```

Comment:
[boulder] There's a lot of logic in here, but we don't have any explicit tests to ensure the integrity. I'd like to see us add a companion test file.

There's a case to be made that this is covered by `TaxAccessStepTwo.test.tsx`, but given the reorganization we should also move related tests to be more domain specific. If moving the tests doesn't make sense, then we could add new tests to just cover what's here.